kratos/utilities/particles_utilities.h

```diff
@@ -0,0 +1,330 @@
+    static void CountParticlesInNodes(
+            BinBasedFastPointLocator<TDim>& rLocator,
+            ModelPart& rVolumeModelPart,
+            const ModelPart& rParticlesModelPart,
+            const Variable<double>& rCounterVariable
+            )
+        {
+
+            //reset the EMBEDDED_VELOCITY
+            block_for_each(rVolumeModelPart.Nodes(), [&rCounterVariable](auto& rNode){
+                rNode.GetValue(rCounterVariable) = 0.0;
+            });
+
+
+            unsigned int max_results = 10000; 
+            typename BinBasedFastPointLocator<TDim>::ResultContainerType TLS(max_results);
+            
+            //for every interface node (nodes in cut elements)
+            block_for_each(rParticlesModelPart.Nodes(), TLS, [&rLocator, &rCounterVariable](const auto& rNode, auto& rTLS){
+
+                Vector shape_functions;
+                Element::Pointer p_element;
+                bool is_found = rLocator.FindPointOnMesh(rNode.Coordinates(), shape_functions, p_element, rTLS.begin(), rTLS.size(), 1e-5);
+
+                if(is_found){
+
+                    auto& r_geom = p_element->GetGeometry();
+                    for(unsigned int i=0; i<r_geom.size(); ++i){
+                        auto& rcounter = r_geom[i].GetValue(rCounterVariable);
+                        AtomicAdd(rcounter, 1.0);
+                    }
+                }
+
+            });
+
+        }
+
+    /**
+    this function looks if particle is found in the locator. If it is not, rVariable is marked with the value "outsider_value"
+    */
+    template<unsigned int TDim, class TScalarType>
+    static void MarkOutsiders(
+            BinBasedFastPointLocator<TDim>& rLocator,
+            ModelPart& rParticlesModelPart,
+            const Variable<TScalarType>& rVariable,
+            const TScalarType outsider_value
+            )
+        {
+            unsigned int max_results = 10000; 
+            typename BinBasedFastPointLocator<TDim>::ResultContainerType TLS(max_results);
+            
+            //for every interface node (nodes in cut elements)
+            block_for_each(rParticlesModelPart.Nodes(), TLS, [&rLocator, &rVariable, outsider_value](auto& rNode, auto& rTLS){
+
+                Vector shape_functions;
+                Element::Pointer p_element;
+                bool is_found = rLocator.FindPointOnMesh(rNode.Coordinates(), shape_functions, p_element, rTLS.begin(), rTLS.size(), 1e-5);
+
+                if(!is_found){
+                    rNode.SetValue(rVariable, outsider_value);
+                }
+            });
+        }
+
+    /**@brief provides the value of the interpolated variable "rInterpolationVariable" at all the positions indicated in rCoordinates
+    */
+    template<unsigned int TDim, class TScalarType>
+    static std::pair< std::vector<bool>, std::vector<TScalarType> > InterpolateValuesAtPoints(
+            BinBasedFastPointLocator<TDim>& rLocator,
+            const Matrix& rCoordinates,
```

Comment:
Wouldn't it be better to pass an `std::vector<array_1d<double,3>>`?  Then the user only needs to `push_back` (or `insert`) the result of calling `Coordinates()`.

Reply:
you would have to export std::vector<array...> to python and to do an interface to accomplish this.

Reply:
It should work, we do that all the time

Reply:
ok, i will try it out

Reply:
it does work (tested) however i would prefer the current version as i can do the following:

~~~py
coords = np.array(self.var_utils.GetInitialPositionsVector(self.all_nodes,3)).reshape(len(self.all_nodes),3)
~~~